Gravity/Gravity.NUnit.Integration/RSAPI_IntegrationTest.cs

```diff
@@ -0,0 +1,395 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using kCura.Relativity.Client;
+using NUnit.Framework;
+using Relativity.API;
+using Relativity.Test.Helpers;
+using Relativity.Test.Helpers.ServiceFactory.Extentions;
+using Relativity.Test.Helpers.SharedTestHelpers;
+using Relativity.Test.Helpers.WorkspaceHelpers;
+using Gravity.Test;
+using System.Configuration;
+using System.Diagnostics.Eventing.Reader;
+using Gravity.Base;
+using Gravity.Test.TestClasses;
+using kCura.Relativity.Client.DTOs;
+using System.Reflection;
+using Gravity.Test.Helpers;
+
+namespace Gravity.NUnit.Integration
+{
+    [TestFixture]
+    public class RSAPI_IntegrationTest
+    {
+        #region Variables
+
+        //this is set to true and should skip the creation of the database, installation of the testing application, and the deletion of the workspace.
+        //must point _workspaceId = valid workspace with application already installed in the Setup
+        private bool _debug = Convert.ToBoolean(ConfigurationManager.AppSettings["Debug"]);
+        private int _debugWorkspaceId = Convert.ToInt32(ConfigurationManager.AppSettings["DebugWorkspaceId"]);
+
+        private IRSAPIClient _client;
+        private readonly string _workspaceName = $"GravityTest_{Guid.NewGuid()}";
+        private int _workspaceId;
+        private IServicesMgr _servicesManager;
+        private IDBContext _eddsDbContext;
+        private IDBContext _dbContext;
+        //public string FilepathApplication = TestHelpers.Constants.Agent.DEFAULT_RAP_FILE_LOCATION + TestHelpers.Constants.Application.General.APPLICATION_NAME;
+        public string _applicaitonFilePath = ConfigurationManager.AppSettings["TestApplicationLocation"];
+        public string _applicationName = ConfigurationManager.AppSettings["TestApplicationName"];
+        private Test.Helpers.TestObjectHelper _testObjectHelper;
+        #endregion
+
+        #region Setup
+        [TestFixtureSetUp]
+        public void Execute_TestFixtureSetup()
+        {
+            try
+            {
+                //Start of test and setup
+                Console.WriteLine("Test START.....");
```

Comment:
These `WriteLine`s add lots of clutter, and should be removed/trimmed to the extent reasonable.

Reply:
In particular, any state management (e.g. "Test is starting") is handled already by the default output of the testing framework.

Reply:
I would like the latter ones in particular implemented before merge.

Reply:
Let me check with a few of the testers.  I think this is required for running automated tests in environments other than the IDE.  I think it makes since what you are saying, I just did this based on recommendations and I will get to the why.

Reply:
One of the biggest advantages of adding any type of logging for tests is that it automatically becomes easily readable to someone who hasn't worked with your code before. A developer is able to skim through the code without having to debug or spend too much time on it. Mark is also right and if you try to run this in any other tool besides Visual Studio, the Writeline statements are the only things that show up and help you debug your tests. We experience this internally when our tests fail in Jenkins. The logging statements help us find the exact location of the failure in jenkins. Hope this helps.